sqlmesh/core/model/definition.py

```diff
@@ -2122,3 +2124,12 @@ def _refs_to_sql(values: t.Any) -> exp.Expression:
     "allow_partials": exp.convert,
     "signals": lambda values: exp.Tuple(expressions=values),
 }
+
+
+def get_model_name(path: Path) -> str:
+    name = path.stem
+    subdirectories = path.parts[path.parts.index("models") + 1 : -1]
```

Comment:
Maybe let's gracefully handle `"models"` not existing in `path.parts`. If for whatever reason that happens, it'll be better UX if we say "If you wanna skip the name in the model, then you need to define it in a file under the project's `models/` directory".

Reply:
I think this is ok since it is checked before. In `/cli/main.py::cli` if the project doesn't have any models:
`if load and not context.models:`
it raises an exception. Though I can add an additional check here as well

Reply:
Btw, I think we can simplify this function by reducing it to:

```python
path_parts = path.parts[...] + [path.stem]
return ".".join(path_parts[-3:]
```